GTHC/web
# Setting Pages v1

Refer to this trello card: https://trello.com/c/CMBzPatl/23-settings-page
## Settings Page
### Purpose
To complete the integration of the Settings page with update API endpoints. Added upon the front end components as well.

### Testing
1. Sign in with a captain account and go to Team Profile
2. Edit Team Profile and save
3. Verify that new info has been saved by checking the Team Profile Pane
4. Do the same in User Profile.
5. Edit password in User Profile (Notice the wall that exists until the correct current password is inputted, verify that the wall is active only when a password has not been submitted or is incorrect) 
6. Verify password has change. This can be done in two ways:
    a. Sign out and in
    b. (recommended) Test through Rails console, which is directly connected to database. Do the following:
```
$ yarn run dev:console
...
irb(main):001:0> User.find_by_email(" <INPUT EMAIL HERE> ").vailid_password? "<INPUT OLD PASSWORD>"
  ...
=> false
irb(main):002:0>User.find_by_email(" <INPUT EMAIL HERE> ").vailid_password? "<INPUT NEW PASSWORD>"
...
=> true
```
If everything is functioning correctly and you replace all of the <> strings in the example above according to your situation, then your output should look similar to above. If not, check your input into the Rails console or this verify's that the password has NOT been changed.

## Demo
![peek 2018-12-21 10-39](https://user-images.githubusercontent.com/13587407/50350438-34cbdc80-050d-11e9-8cbb-ba4e832bf2a0.gif)
